forms/models/elements/subform.js

```diff
@@ -256,7 +256,14 @@ define(function (require) {
 
       return new Promise(function (resolve, reject) {
         Promise.all(promises).then(function (values) {
-          resolve(values);
+          values = _.reduce(values, function (memo, value) {
+            if (value._action === 'remove' && (value.id === null || value.id === undefined)) {
```

Comment:
MySQL is not the only possible destination for this data. I agree that truthy / falsey tests are inadequate here, but I can't rule out that this isn't being sent to a custom web service that requires ID:0 in order to function.

We should try to be as specific as we can whilst avoiding compatibility issues. I would only rule out ID:0 here if we knew that the query builder would do bad things with that. Otherwise, I think any non-empty String or any Number should be fine here.